src/lib/types/WebSearch.ts

```diff
@@ -14,33 +15,31 @@ export interface WebSearch extends Timestamps {
 }
 
 export interface WebSearchSource {
-	title: string;
+	title?: string;
 	link: string;
-	hostname: string;
-	text?: string; // You.com provides text of webpage right away
+}
+export interface WebSearchScrapedSource extends WebSearchSource {
+	page: WebSearchPage;
+}
+export interface WebSearchPage {
+	title: string;
+	siteName?: string;
+	author?: string;
+	description?: string;
+	createdAt?: string;
+	modifiedAt?: string;
```

Comment:
they should be of type: `Date`

Reply:
actually, you can make WebSearchPage extends

https://github.com/huggingface/chat-ui/blob/492c51fff022ba40410056c0889b974f14a83471/src/lib/types/Timestamps.ts#L1-L4

and delete fields `createdAt` & `modifiedAt`

Reply:
They're actually strings because we don't know what format the dates will be in when we parse them from the page. For now we just grab the value in the meta elems

https://github.com/huggingface/chat-ui/pull/1094/files#diff-d4fd570f6480c02bb93d280a06da3a741b497b41c9878984197a3f9ec005030eR522-R528